### PR TITLE
Add Nodes Pages

### PR DESCRIPTION
**Pull request by BramGruneir**
This adds single and all node pages to the monitoring ui.  
Just throwing it out there to make sure the architecting is correct.

@MattTracy please take a look.

### CODE REVIEW

File: resource/resource.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ package resource
 // Be sure to remove this flag and go generate before creating a PR. Also, make
 // sure you clear the page cache when debugging or you might not see the
 // changes.
-//go:generate go-bindata -pkg resource -mode 0644 -modtime 1400000000 -o ./embedded.go ./ui/...
+//go:generate go-bindata -debug -pkg resource -mode 0644 -modtime 1400000000 -o ./embedded.go ./ui/...
```
Comment (BramGruneir) [inline resource/resource.go]: Obviously, I'll remove the debug when I'm done testing.

File: resource/ui/ts/pages/nodes.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          ])
+        ]);
+      }
+    }
+
+    export module SinglePage {
+      export function controller() { }
+      export function view() {
+        var nodeId = m.route.param("node_id");
+        Model.nodeStatuses.QueryOne(nodeId)
```
Comment (mrtracy) [inline resource/ui/ts/pages/nodes.ts]: This is the biggest issue - you should never issue queries in a view.  Each "view" method is called after every single data change that mithril knows about - that is, any completed network request (if called through m.request) or after any event handler hooked up via an m() object.

View methods need to be cheap, and should basically only be generating a set of m() objects.  Querying should be managed in the controller.

File: resource/ui/ts/models/node_status.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                        return results;
+                    });
+            }
+
+            QueryOne(nodeId:string): _mithril.MithrilPromise<INodeStatus> {
+                var url = "/_status/nodes/" + nodeId;
+                return m.request({ url: url, method: "GET", extract: nonJsonErrors })
+                    .then((status: INodeStatus) => {
+                        if (this._data()[status.desc.node_id] == null) {
+                            this._data()[status.desc.node_id] = <Models.NodeStatus.INodeStatus[]>[];
```
Comment (mrtracy) [inline resource/ui/ts/models/node_status.ts]: I don't think you need to cast here, the statement already has type information from the... is it still lvalue in javascript?  The variable being assigned has a type.

File: resource/ui/ts/pages/nodes.ts (hunk 2 above)
Comment (BramGruneir) [inline resource/ui/ts/pages/nodes.ts]: Fixed.

**Comment by BramGruneir** [comment]
@mrtracy PTAL

File: resource/ui/ts/models/node_status.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                    });
+            }
+
+            private _updateDescriptions(): void {
+                this.desc(<NodeDescriptionMap> {});
+                var nodeId: string;
+                for (nodeId in this._data()) {
+                    this.desc()[nodeId] = this._data()[nodeId][this._data()[nodeId].length - 1].desc;
+                }
+                return;
```
Comment (mrtracy) [inline resource/ui/ts/models/node_status.ts]: return statement not needed

File: resource/ui/ts/pages/nodes.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+      export function view(ctrl: Controller) {
+        return m("div", [
+          m("h2", "Nodes Status"),
+          m("div", [
+            m("h3", "Nodes"),
+            m("ul", [
+              Object.keys(nodeStatuses.desc()).sort().map(function(nodeId) {
+                var desc = nodeStatuses.desc()[nodeId];
+                return m("li", { key: desc.node_id },
+                  m("a[href=/nodes/" + nodeId + "]",
```
Comment (mrtracy) [inline resource/ui/ts/pages/nodes.ts]: missing closing bracket on the selector string, should be `"a[href=/nodes/"]`

**Comment by mrtracy** [comment]
LGTM

**Pull request merged**